src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs

```diff
@@ -980,5 +980,63 @@ public void NominalRecordMembers()
                 "C.C()",
             }, members.Select(m => m.ToTestDisplayString(includeNonNullable: true)));
         }
+
+        [Fact]
+        public void PartialTypes_01()
+        {
+            var src = @"
+using System;
+partial record C(int X, int Y)
+{
+    public static void Main()
+    {
+        var c = new C(1, 2);
+        Console.WriteLine(c.X);
+        Console.WriteLine(c.Y);
+    }
+}
+
+partial record C(int X, int Y)
+{
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (13,17): error CS8863: Only a single record partial declaration may have a parameter list
+                // partial record C(int X, int Y)
+                Diagnostic(ErrorCode.ERR_MultipleRecordParameterLists, "(int X, int Y)").WithLocation(13, 17)
+                );
+
+            Assert.Equal(new[] { "C..ctor(System.Int32 X, System.Int32 Y)", "C..ctor(C )" }, comp.GetTypeByMetadataName("C")!.Constructors.Select(m => m.ToTestDisplayString()));
+        }
+
+        [Fact]
+        public void PartialTypes_02()
```

Comment:
>PartialTypes_02 [](start = 20, length = 15)

I think we would want at least one success test. Let's say 3 declarations with initializers and the one in the middle has parameters. Should verify that all members are properly initialized and observe IL to verify order of initialization